server/src/main/java/org/elasticsearch/search/aggregations/pipeline/movfn/MovingFunctions.java

```diff
@@ -21,59 +21,83 @@
 
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.function.ToDoubleFunction;
 
 /**
  * Provides a collection of static utility methods that can be referenced from MovingFunction script contexts
  */
 public class MovingFunctions {
 
     /**
-     * Find the maximum value in a window of values
+     * Find the maximum value in a window of values.
+     * If all values are missing/null/NaN, the return value will be NaN
      */
     public static double windowMax(Collection<Double> values) {
         return values.stream().max(Double::compareTo).orElse(Double.NaN);
     }
 
     /**
      * Find the minimum value in a window of values
+     * If all values are missing/null/NaN, the return value will be NaN
      */
     public static double windowMin(Collection<Double> values) {
         return values.stream().min(Double::compareTo).orElse(Double.NaN);
     }
 
     /**
      * Find the sum of a window of values
+     * If all values are missing/null/NaN, the return value will be 0.0
      */
     public static double windowSum(Collection<Double> values) {
-        return values.stream().mapToDouble(Double::doubleValue).sum();
+        if (values.size() == 0) {
+            return 0.0;
```

Comment:
From the docs change above it says:
> If the window is empty, or all values are `null`/`NaN`, `0.0` is returned as the result.

This seems to not be the case here?

Reply:
?  If the size is zero, it returns `0.0`.  And in the `mapToDouble()`, we only use the value if it's non-null and non-NaN (otherwise we return 0.0 for that position).

There's a test to verify this: https://github.com/elastic/elasticsearch/pull/29594/files/83778999a9aec1e0170efba0d4857100a927a770..c93d065b6ffd4e37089b4e251ad784380b75a942#diff-a35897b4e9c0a693a4515cab553a6427R73